dockerfiles/sccache-ci-linux/Dockerfile

```diff
@@ -0,0 +1,85 @@
+FROM ubuntu:groovy
+
+ARG VCS_REF=master
+ARG BUILD_DATE=""
+ARG REGISTRY_PATH=paritytech
+
+# metadata
+LABEL summary="CI image with all dependencies for sccache compilation." \
+	name="${REGISTRY_PATH}/sccache-ci-linux" \
+	maintainer="devops-team@parity.io" \
+	version="1.0" \
+	description="libssl-dev, clang, libclang-dev, lld, cmake, make, git, pkg-config \
+curl, time, rhash, rust stable, rust nightly, sccache" \
+	io.parity.image.vendor="Parity Technologies" \
+	io.parity.image.source="https://github.com/paritytech/scripts/blob/${VCS_REF}/\
+dockerfiles/sccache-ci-linux/Dockerfile" \
+	io.parity.image.documentation="https://github.com/paritytech/scripts/blob/${VCS_REF}/\
+dockerfiles/sccache-ci-linux/README.md" \
+	io.parity.image.revision="${VCS_REF}" \
+	io.parity.image.created="${BUILD_DATE}"
+
+WORKDIR /builds
+
+# config for wasm32-unknown-unknown & clang
+COPY utility/base-ci-linux-config /root/.cargo/config
+
+ENV RUSTUP_HOME=/usr/local/rustup \
+	CARGO_HOME=/usr/local/cargo \
+	PATH=/usr/local/cargo/bin:/usr/local/cuda-11.1/bin:$PATH \
+	CC=clang \
+	CXX=clang \
+    LD_LIBRARY_PATH=/usr/local/cuda-11.1/lib64 \
+    DEBIAN_FRONTEND=noninteractive \
+    TERM=xterm-256color
+
+# install tools and dependencies
+RUN set -eux; \
```

Comment:
With newer versions of podman / docker should be able to fuse the layers, so the spaghetti/chain wars should not be necessary anymore but changes can take advantage of cached layers :)

Reply:
Yeah, the thing is we ignore this layer cache on the CI images builds.
And the reason is I didn't update myself about some bugs that were in docker layer caching (like if the line in Dockerfile that `COPY`es the file from local doesn't change, but the file occasionally got changed, still the old file version can be taken from the cache).